docs/searching.rst

```diff
@@ -0,0 +1,348 @@
+Searching
+=========
+
+Almost every interaction with the Carbon Black Cloud SDK will involve searching for some object on the server that your
+code can then inspect or operate on.  Searching in the SDK involves three steps:
+
+1. Create a *query object* with the ``select()`` method.
+2. Refine the query by using the query object's methods to add a text *query* and/or *search criteria.*
+3. Execute the query to see its results.
+
+Creating a Query Object
+-----------------------
+
+A query object is created via the :func:`CBCloudAPI.select() <cbc_sdk.connection.BaseAPI.select>` operation, specifying
+the type of data to be retrieved.
+
+In this example, we create a query to search for all devices with antivirus active::
+
+    # assume the CBCloudAPI object is in the variable "api"
+    >>> from cbc_sdk.platform import Device
+    >>> device_query = api.select(Device).where('avStatus:AV_ACTIVE')
+
+    # The device query has been created but not yet executed
+    >>> type(device_query)
+    <class 'cbc_sdk.platform.devices.DeviceSearchQuery'>
+
+The ``select()`` method may take either a class reference or a string class name::
+
+    >>> query1 = api.select(Device)
+    >>> query2 = api.select("Device")
+
+    # prove that the query we get back in either case is the same
+    >>> type(query1) == type(query2)
+    True
+
+Selecting an Object Directly
+****************************
+
+The ``select()`` method may also be used to retrieve an object directly if you know its ID value, by passing the ID as
+a second parameter::
+
+    >>> dev = api.select(Device, 1234567)  # assume this device exists
+    >>> type(dev)
+    <class 'cbc_sdk.platform.devices.Device'>
+
+Refining a Query
+----------------
+
+Queries may support one of two different methods for refining a query:
+
+* Through the use of *text query.*
+* Through adding *criteria.*
+
+Text Query Support
+******************
+
+Text queries may be added to a query object by using the query object's
+:func:`where() <cbc_sdk.base.QueryBuilder.where>`, :func:`and_() <cbc_sdk.base.QueryBuilder.and_>`, and
+:func:`or_() <cbc_sdk.base.QueryBuilder.or_>` methods.  The following example sets up a query looking for events
+in which the program ``googleupdate.exe`` accesses the system registry on a device with a specific hostname, IP
+address, and owner::
+
+    # assume the CBCloudAPI object is in the variable "api"
+    from cbc_sdk.endpoint_standard import Event
+    >>> event_query = api.select(Event).where(hostName='Win10').and_(ipAddress='10.0.0.1')
```

Comment:
It seems odd to use an example with camelCase properties and uses the old integrationService API

Reply:
I didn't change the example that was present in the old "concepts" document. If you've got a better idea for this, I'm willing to hear it. Preferably with example code I can just drop in in its place. 😃

Reply:
I'd like you come up with a new example based on what would be most useful to a new user. Alerts or Observations would be to potential options

Reply:
And I am _asking_ for _help,_ because I don't know either of those as well as you do.

Reply:
Make an alert search request with a `parent_name` and `process_name`. Look for an interesting alert in the prod02 partner org and take it's parent_name and process_name. You could even use the upcoming new `Alert` model name too

Reply:
I am replacing this with an Observation query, because if I try to run an Alert query, it'll be with Alerts v6 and I think you might object if it weren't with Alerts v7. Observation is safer. And this is just an example of query-building syntax anyway.